### PR TITLE
neovim: update to 0.10.2

### PR DESCRIPTION
**Pull request by Suyun114**
Topic Description
-----------------

- neovim: update to 0.10.2
    Co-authored-by: Suyun (@Suyun114)

Package(s) Affected
-------------------

- neovim: 1:0.10.2

Security Update?
----------------

No

Build Order
-----------

```
#buildit tree-sitter tree-sitter-c tree-sitter-lua tree-sitter-markdown tree-sitter-query tree-sitter-vim tree-sitter-vimdoc neovim
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`

### CODE REVIEW

**Comment by Suyun114** [comment]
Tested on amd64.

image

**Comment by xtexx** [comment]
Tested-by: ~~~ # AMD64

**Review by xtexx — DISMISSED** [review]

**Comment by OriginCode** [comment]
LGTM on amd64
圖片

File: app-editors/neovim/autobuild/defines [hunk 1]
```diff
@@ -4,7 +4,7 @@ PKGDES="Vim-fork focused on extensibility and usability"
 
 PKGEPOCH=1
 
-PKGDEP="luajit msgpack-c tree-sitter unibilium libvterm libuv libtermkey libluv lpeg"
+PKGDEP="luajit msgpack-c tree-sitter tree-sitter-c tree-sitter-lua tree-sitter-markdown tree-sitter-query tree-sitter-vim tree-sitter-vimdoc unibilium libvterm libuv libtermkey libluv lpeg"
```
Comment (OriginCode) [inline app-editors/neovim/autobuild/defines]: It's easier to read if you break this long line at a 80 col limit.

**Review by OriginCode — APPROVED** [review]

**Pull request merged**